src/cursor/abstract_cursor.ts

```diff
@@ -767,15 +785,27 @@ export abstract class AbstractCursor<
         'Unexpected null selectedServer. A cursor creating command should have set this'
       );
     }
+    const getMoreOptions = {
+      ...this.cursorOptions,
+      session: this.cursorSession,
+      batchSize,
+      omitMaxTimeMS:
+        this.cursorOptions.omitMaxTimeMS || this.cursorOptions.useMaxAwaitTimeMSAsMaxTimeMS
+    };
+
+    if (
+      this.cursorOptions.tailable &&
+      this.cursorOptions.awaitData &&
+      this.cursorOptions.maxAwaitTimeMS != null
+    ) {
+      getMoreOptions.maxTimeMS = this.cursorOptions.maxAwaitTimeMS;
```

Comment:
This does nothing - `GetMoreOperation.execute()` already conditionally attaches a maxTimeMS equal to `maxAwaitTimeMS`

Reply:
Good catch here